crates/router/src/core/payouts.rs

```diff
@@ -1980,6 +2062,37 @@ pub async fn payout_create_db_entries(
             })
         })?
         .customer_id;
+    let payout_link = if let Some(payout_link_create) = req.payout_link {
+        if payout_link_create {
+            Some(
+                validator::create_payout_link(
+                    state,
+                    merchant_account,
+                    &req.payout_link_config,
+                    &customer_id,
+                    req.return_url.to_owned(),
+                    payout_id,
+                    req.amount
+                        .as_ref()
+                        .get_required_value("amount")
+                        .attach_printable(
+                            "amount is a required value when creating payout links",
+                        )?,
+                    req.currency
+                        .as_ref()
+                        .get_required_value("currency")
+                        .attach_printable(
+                            "currency is a required value when creating payout links",
+                        )?,
+                )
+                .await?,
+            )
+        } else {
+            None
+        }
+    } else {
+        None
+    };
```

Comment:
can we do something like


```suggestion
   let payout_link = match req.payout_link {
    Some(true) => {
        let amount = req.amount
            .as_ref()
            .get_required_value("amount")
            .attach_printable("amount is a required value when creating payout links")?;
            
        let currency = req.currency
            .as_ref()
            .get_required_value("currency")
            .attach_printable("currency is a required value when creating payout links")?;
        let payout_link_config = req.payout_link_config.as_ref().ok_or_else(|| {
            report!(errors::ApiErrorResponse::MissingRequiredField {
                field_name: "payout_link_config",
            })
        })?;
        Some(
            validator::create_payout_link(
                state,
                merchant_account,
                payout_link_config,
                &customer_id,
                req.return_url.to_owned(),
                payout_id,
                amount,
                currency,
            )
            .await?,
        )
    },
    _ => None,
};
```